cryptography.py

```diff
@@ -121,48 +121,98 @@ def tonelli_shanks(n: int, p: int):
 
 
 '''Cryptographic Keys'''
+'''
+For RSA, we don't have a way of retrieving the public key from the private key.
+Further, we don't have a way to get p and q from n without factoring the number.
+Hence given a random number s, we find the first prime above and below this number,
+    and e will be the greatest number lower than s satisfying (e, phi(n)) = 1
+
+'''
+
+
+def generate_above_below_primes(midpoint: int):
```

Comment:
This method is probably not secure. Given n, it reduces the factoring problem by only needing to search within the square root of n. It would be better to create a method to generate random primes based on a seed value. Then get p and q using this method.